internal/provider/resource_kubernetes_cloud.go

```diff
@@ -82,17 +116,146 @@ func (o *kubernetesCloudResource) Schema(_ context.Context, req resource.SchemaR
 }
 
 // Create adds a new kubernetes cloud to controllers used now by Terraform provider.
-func (o *kubernetesCloudResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+func (r *kubernetesCloudResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Prevent panic if the provider has not been configured.
+	if r.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "kubernetes_cloud", "create")
+		return
+	}
+
+	var plan kubernetesCloudResourceModel
+
+	// Read Terraform configuration from the request into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create the kubernetes cloud.
+	cloudCredentialName, err := r.client.Clouds.CreateKubernetesCloud(
+		&juju.CreateKubernetesCloudInput{
+			Name:             plan.CloudName.ValueString(),
+			KubernetesConfig: plan.KubernetesConfig.ValueString(),
+		},
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create kubernetes cloud, got error %s", err))
+		return
+	}
+
+	r.trace(fmt.Sprintf("Created kubernetes cloud %s", plan.CloudName.ValueString()))
+
+	plan.ID = types.StringValue(newKubernetesCloudID(plan.CloudName.ValueString()))
```

Comment:
We may want to update this to include the credential name at some point too for Import.